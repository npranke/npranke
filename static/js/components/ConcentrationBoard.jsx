import PropTypes from 'prop-types'
import React from 'react'

import ConcentrationBoardRow from './ConcentrationBoardRow'

export function ConcentrationBoard(props) {
    const boardPictures = props.boardOrder.map((pictureName) => {
        return ({ [pictureName]: props.pictures[pictureName] })
    })

    return (
        props.isPortrait
            ? (
                <div className="board">
                    <div className="board-row-wrapper">
                        <ConcentrationBoardRow
                            pictures={ boardPictures.slice(0, 4) }
                            matches={ props.matches }
                            first={ props.first }
                            second={ props.second }
                        />
                    </div>
                    <div className="board-row-wrapper">
                        <ConcentrationBoardRow
                            pictures={ boardPictures.slice(4, 8) }
                            matches={ props.matches }
                            first={ props.first }
                            second={ props.second }
                        />
                    </div>
                    <div className="board-row-wrapper">
                        <ConcentrationBoardRow
                            pictures={ boardPictures.slice(8, 12) }
                            matches={ props.matches }
                            first={ props.first }
                            second={ props.second }
                        />
                    </div>
                    <div className="board-row-wrapper">
                        <ConcentrationBoardRow
                            pictures={ boardPictures.slice(12, 16) }
                            matches={ props.matches }
                            first={ props.first }
                            second={ props.second }
                        />
                    </div>
                    <div className="board-row-wrapper">
                        <ConcentrationBoardRow
                            pictures={ boardPictures.slice(16, 20) }
                            matches={ props.matches }
                            first={ props.first }
                            second={ props.second }
                        />
                    </div>
                    <div className="board-row-wrapper">
                        <ConcentrationBoardRow
                            pictures={ boardPictures.slice(20, 24) }
                            matches={ props.matches }
                            first={ props.first }
                            second={ props.second }
                        />
                    </div>
                </div>
            ) : (
                <div className="board">
                    <div className="board-row-wrapper">
                        <ConcentrationBoardRow
                            pictures={ boardPictures.slice(0, 6) }
                            matches={ props.matches }
                            first={ props.first }
                            second={ props.second }
                        />
                    </div>
                    <div className="board-row-wrapper">
                        <ConcentrationBoardRow
                            pictures={ boardPictures.slice(6, 12) }
                            matches={ props.matches }
                            first={ props.first }
                            second={ props.second }
                        />
                    </div>
                    <div className="board-row-wrapper">
                        <ConcentrationBoardRow
                            pictures={ boardPictures.slice(12, 18) }
                            matches={ props.matches }
                            first={ props.first }
                            second={ props.second }
                        />
                    </div>
                    <div className="board-row-wrapper">
                        <ConcentrationBoardRow
                            pictures={ boardPictures.slice(18, 24) }
                            matches={ props.matches }
                            first={ props.first }
                            second={ props.second }
                        />
                    </div>
                </div>
            )
    )
}

ConcentrationBoard.defaultProps = {
    first: {
        id: null,
        pictureid: null,
    },
    isPortrait: false,
    matches: [],
    second: {
        id: null,
        pictureid: null,
    },
}

ConcentrationBoard.propTypes = {
    boardOrder: PropTypes.arrayOf(PropTypes.string).isRequired,
    first: PropTypes.shape({
        id: PropTypes.string,
        pictureid: PropTypes.string,
    }),
    isPortrait: PropTypes.bool,
    matches: PropTypes.arrayOf(PropTypes.string),
    pictures: PropTypes.objectOf(
        PropTypes.shape({
            id: PropTypes.string.isRequired,
            pictureid: PropTypes.string.isRequired,
            back: PropTypes.node.isRequired,
            front: PropTypes.node.isRequired,
            matched: PropTypes.node.isRequired,
        }).isRequired,
    ).isRequired,
    second: PropTypes.shape({
        id: PropTypes.string,
        pictureid: PropTypes.string,
    }),
}

export default ConcentrationBoard
