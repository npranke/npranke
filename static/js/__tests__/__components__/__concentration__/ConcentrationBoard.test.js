import { render, screen } from '@testing-library/react'

import ConcentrationBoard from '@components/concentration/ConcentrationBoard'

const pictures = Array.from({ length: 12 }, (value, index) => {
    return `${index}`
}).reduce((pictureAccumulator, pictureid) => {
    pictureAccumulator[`picture${pictureid}-a`] = {
        id: `${pictureid}-a`,
        pictureid,
        back: <img src="" alt="" />,
        front: <img src="" alt="" />,
        matched: <img src="" alt="" />,
    }

    pictureAccumulator[`picture${pictureid}-b`] = {
        id: `${pictureid}-b`,
        pictureid,
        back: <img src="" alt="" />,
        front: <img src="" alt="" />,
        matched: <img src="" alt="" />,
    }

    return pictureAccumulator
}, {})

const boardOrder = Array.from({ length: 12 }, (value, index) => {
    return `picture${index}-a`
}).concat(Array.from({ length: 12 }, (value, index) => {
    return `picture${index}-b`
}))

describe('ConcentrationBoard', () => {
    test('has 24 images when not isPortrait', () => {
        render(
            <ConcentrationBoard
                pictures={ pictures }
                boardOrder={ boardOrder }
            />,
        )

        expect(
            screen.getAllByRole('img'),
        ).toHaveLength(24)
    })

    test('has 24 images when isPortrait', () => {
        render(
            <ConcentrationBoard
                isPortrait
                pictures={ pictures }
                boardOrder={ boardOrder }
            />,
        )

        expect(
            screen.getAllByRole('img'),
        ).toHaveLength(24)
    })
})

describe('ConcentrationBoard snapshot', () => {
    test('matches snapshot when not isPortrait', () => {
        const { asFragment } = render(
            <ConcentrationBoard
                pictures={ pictures }
                boardOrder={ boardOrder }
            />,
        )

        expect(asFragment()).toMatchSnapshot()
    })

    test('matches snapshot when isPortrait', () => {
        const { asFragment } = render(
            <ConcentrationBoard
                isPortrait
                pictures={ pictures }
                boardOrder={ boardOrder }
            />,
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
