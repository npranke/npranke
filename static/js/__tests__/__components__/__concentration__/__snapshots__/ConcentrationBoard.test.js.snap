// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConcentrationBoard snapshot matches snapshot when isPortrait 1`] = `
<DocumentFragment>
  <div
    class="board"
  >
    <div
      class="board-row-wrapper"
    >
      <div
        class="board-row"
      >
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
      </div>
    </div>
    <div
      class="board-row-wrapper"
    >
      <div
        class="board-row"
      >
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
      </div>
    </div>
    <div
      class="board-row-wrapper"
    >
      <div
        class="board-row"
      >
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
      </div>
    </div>
    <div
      class="board-row-wrapper"
    >
      <div
        class="board-row"
      >
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
      </div>
    </div>
    <div
      class="board-row-wrapper"
    >
      <div
        class="board-row"
      >
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
      </div>
    </div>
    <div
      class="board-row-wrapper"
    >
      <div
        class="board-row"
      >
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ConcentrationBoard snapshot matches snapshot when not isPortrait 1`] = `
<DocumentFragment>
  <div
    class="board"
  >
    <div
      class="board-row-wrapper"
    >
      <div
        class="board-row"
      >
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
      </div>
    </div>
    <div
      class="board-row-wrapper"
    >
      <div
        class="board-row"
      >
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
      </div>
    </div>
    <div
      class="board-row-wrapper"
    >
      <div
        class="board-row"
      >
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
      </div>
    </div>
    <div
      class="board-row-wrapper"
    >
      <div
        class="board-row"
      >
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
        <img
          alt=""
          src=""
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
